pkg/controller/template/render_test.go

```diff
@@ -48,11 +45,26 @@ func TestCloudProvider(t *testing.T) {
 	}, {
 		platform:    configv1.GCPPlatformType,
 		featureGate: newFeatures("cluster", "CustomNoUpgrade", []string{cloudprovider.ExternalCloudProviderFeature}, nil),
-		res:         "gce",
+		res:         "external",
+	}, {
+		platform:    configv1.VSpherePlatformType,
+		featureGate: newFeatures("cluster", "CustomNoUpgrade", []string{cloudprovider.ExternalCloudProviderFeature}, nil),
+		res:         "external",
+	}, {
+		platform:    configv1.OpenStackPlatformType,
+		featureGate: newFeatures("cluster", "CustomNoUpgrade", []string{cloudprovider.ExternalCloudProviderFeature}, nil),
+		res:         "external",
+	}, {
+		platform:    configv1.OpenStackPlatformType,
+		featureGate: newFeatures("cluster", "CustomNoUpgrade", []string{cloudprovider.ExternalCloudProviderFeature}, nil),
+		res:         "external",
```

Comment:
Is there a reason why these two OpenStack cases seem to be identical? Looks like an accidental dupe?

Reply:
Yup, accidental. Fixed.